### PR TITLE
Initial drop of componenttest public API and provider API source

### LINKED ISSUE #20485: Check in and simplify componenttest/simplicity source
https://github.com/OpenLiberty/open-liberty/pull/20396

### PR DESCRIPTION
**Pull request by tevans78**
I am removing the binary dependency on a number of `com.ibm.ws.componenttest` artifacts by checking in the source for some and stripping out what we are not using. This is just the first step of what I want to do. Once the OL side is using just source it will be easier to refactor and make the code common with what is used in the WL side. There are a lot of APIs here that are clearly just from tWAS and I don’t think there is any benefit to keeping them for use in Liberty.

Converted to source (merged into fattest.simplicity)
- public.api
- fat.util

Completely removed
- provider.api
- com.ibm.componenttest.common
- com.ibm.ws.topology.helper
- fat.harness

### CODE REVIEW

**Comment by tevans78** [comment]
#libby

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
PB: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_CR5CoKVEEeyLpORmV0nVgA

File: dev/com.ibm.ws.common.encoder/bnd.bnd [hunk 1]
```diff
@@ -25,9 +25,8 @@ publish.wlp.jar.disabled: true
 -testpath: \
 	../build.sharedResources/lib/junit/old/junit.jar;version=file, \
 	com.ibm.ws.junit.extensions;version=latest, \
-	org.hamcrest:hamcrest-all;version=1.3, \
-	org.jmock:jmock-junit4;strategy=exact;version=2.5.1, \
-	org.jmock:jmock;strategy=exact;version=2.5.1, \
```
Comment (jhanders34) [inline dev/com.ibm.ws.common.encoder/bnd.bnd]: This change seemed out of place and unrelated.  What led to doing this change?

File: dev/fattest.simplicity/bnd.bnd [hunk 2]
```diff
@@ -9,48 +9,60 @@
...
+	com.ibm.websphere.simplicity.commands.*,\
+	com.ibm.websphere.simplicity.configuration.*,\
+	com.ibm.websphere.simplicity.exception.*,\
+	com.ibm.websphere.simplicity.jobmanager.*,\
+	com.ibm.websphere.simplicity.product.*,\
+	com.ibm.websphere.simplicity.provider.*,\
+	com.ibm.websphere.simplicity.runtime.*,\
+	com.ibm.websphere.simplicity.server.*,\
+	com.ibm.websphere.simplicity.util.*,\
+	com.ibm.websphere.simplicity.wsadmin.*
```
Comment (jhanders34) [inline dev/fattest.simplicity/bnd.bnd]:
```suggestion
	com.ibm.websphere.simplicity.configuration
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+/**
+ * This is a general container class to hold connection information. This information can be used to
+ * connect to entities suchs as a WebSphere cell or a remote machine.
+ */
+public class ConnectionInfo implements Cloneable {
+	
+	private static Class c = ConnectionInfo.class;
+
+	protected String profileDir;
+    protected ConnectorType connType;
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+     */
+    public ConnectionInfo(ConnectorType connType, String host, Integer port, String user, String password) {
+    	Log.entering(c, "<clinit>", new Object[] { connType, host, port, user, password });
+        this.connType = connType;
+        this.host = host;
+        this.port = port;
+        this.user = user;
+        this.password = password;
+        Log.exiting(c, "<clinit>");
+    }
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    public ConnectionInfo(ConnectorType connType, String host, Integer port, String user, String password, String profileDir) {
+    	Log.entering(c, "<clinit>", new Object[] { connType, host, port, user, password, profileDir });
+        this.connType = connType;
+        this.host = host;
+        this.port = port;
+        this.user = user;
+        this.password = password;
+        this.profileDir = profileDir.replace('\\', '/');
+        Log.exiting(c, "<clinit>");
+    }
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+
+    /**
+     * Set the WebSphere {@link ConnectorType}
+     *  
+     * @param connType
+     */
+    void setConnType(ConnectorType connType) {
+        this.connType = connType;
+    }
+
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+        if(o == null) {
+            return false;
+        }
+        if(!(o instanceof ConnectionInfo)) {
+            return false;
+        }
+        
+        ConnectionInfo other = (ConnectionInfo)o;
+        if(((this.connType == null) && (other.connType != null)) || ((this.connType != null) && (this.connType != other.connType))
+                || ((this.host == null) && (other.host != null)) || ((this.host != null) && (!this.host.equalsIgnoreCase(other.host)))
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
        if(((this.host == null) && (other.host != null)) || ((this.host != null) && (!this.host.equalsIgnoreCase(other.host)))
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+     * @param port The port to connect to
+     * @param user The user name to use when connecting
+     * @param password The password to use when connecting
+     * @return Cloned ConnectionInfo
+     * @throws CloneNotSupportedException
+     */
+    public Object clone(ConnectorType connType, int port, String user, String password) throws CloneNotSupportedException {
+    	ConnectionInfo nci = (ConnectionInfo)this.clone();
+    	if (connType != null)
+    		nci.setConnType(connType);
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
     * @param port The port to connect to
     * @param user The user name to use when connecting
     * @param password The password to use when connecting
     * @return Cloned ConnectionInfo
     * @throws CloneNotSupportedException
     */
    public Object clone(int port, String user, String password) throws CloneNotSupportedException {
    	ConnectionInfo nci = (ConnectionInfo)this.clone();
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/LocalFile.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+package com.ibm.websphere.simplicity;
+
+import java.io.File;
+
+/**
+ * Represents a file or directory on the local {@link Machine}. This file or directory may or may not
+ * actually exist on the local {@link Machine}
+ */
+public class LocalFile extends RemoteFile {
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/LocalFile.java:20]: Keep this file.

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java [hunk 10]
```diff
@@ -0,0 +1,297 @@
+/*******************************************************************************
+ * Copyright (c) 2011 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ConnectionInfo.java]:
```suggestion
 * Copyright (c) 2011,2022 IBM Corporation and others.
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/Machine.java [hunk 11]
```diff
@@ -22,6 +22,7 @@
 import com.ibm.websphere.simplicity.provider.commandline.CommandLineProvider;
 import com.ibm.websphere.simplicity.provider.commandline.RemoteCommandFactory;
 import com.ibm.websphere.simplicity.provider.commandline.local.LocalCommandLineProvider;
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/Machine.java]:
```suggestion
```

And remove the `RemoteCommand executeAsync(String cmd, String[] parameters, String workDir, Properties envVars) throws Exception` method

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ProgramOutput.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.websphere.simplicity;
+
+/**
+ * This class is a container to hold output information for command line command results
+ */
+public class ProgramOutput {
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ProgramOutput.java:17]: Keep this file.

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/ConfigurationProvider.java [hunk 13]
```diff
@@ -0,0 +1,175 @@
+/*******************************************************************************
+ * Copyright (c) 2011 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/ConfigurationProvider.java]:
```suggestion
 * Copyright (c) 2011,2022 IBM Corporation and others.
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/ConfigurationProvider.java [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+			if (ext.equalsIgnoreCase("XML"))
+				ret = new XmlConfigurationProvider(configFile);
+			else if (ext.equalsIgnoreCase("props") || ext.equalsIgnoreCase("properties"))
+				ret = new PropertiesConfigurationProvider(configFile);
+		}
+		
+		Log.exiting(c, "getProvider", ret);
+		return ret;
+	}
+
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/ConfigurationProvider.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/ConfigurationProvider.java [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Properties;
+
+import com.ibm.websphere.simplicity.log.Log;
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/ConfigurationProvider.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/FormattedProperties.java [hunk 16]
```diff
@@ -0,0 +1,196 @@
+/*******************************************************************************
+ * Copyright (c) 2011 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/FormattedProperties.java]:
```suggestion
 * Copyright (c) 2011,2022 IBM Corporation and others.
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/FormattedProperties.java [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import com.ibm.websphere.simplicity.BootStrappingProperty;
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/FormattedProperties.java]:
```suggestion
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/FormattedProperties.java [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+
+        userProps = new HashMap<String, String>();
+        generatedProps = new HashMap<String, String>();
+
+        Set<Object> props = super.keySet();
+        String property = null;
+        for (Object prop : props) {
+            property = (String) prop;
+            if (getProperty(property) != null) {
+                if (property.indexOf(BootStrappingProperty.DATA.toString()) == -1) {
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/FormattedProperties.java]:
```suggestion
                if (property.indexOf("data") == -1) {
```

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/PropertiesConfigurationProvider.java [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+import java.io.OutputStream;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * A {@link ConfigurationProvider} for a properties configuration file. This provider uses the
+ * {@link FormattedProperties} Object to manage and write properties.
+ */
+public class PropertiesConfigurationProvider extends ConfigurationProvider {
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/configuration/PropertiesConfigurationProvider.java:24]: Keep this file.

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/log/Log.java [hunk 20]
```diff
@@ -0,0 +1,421 @@
...
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+public class Log {
```
Comment (jhanders34) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/log/Log.java:33]: Keep this file.

**Review by jhanders34 — CHANGES_REQUESTED** [review]
I do not want to add 1400+ files to fattest.simplicity when we can do so much less.  In the end with some cleanup of existing fattest.simplicity classes and removal of some unused methods, you only need 7 classes instead of 1400+.  I have added the necessary changes in the review comments with suggestions to make it easy to see what needs to be changed.  If I did not comment on a class, that class can be removed.

The only other change that needs to be made is in the com.ibm.ws.fat.util.Props class:

`properties.put("simplicity.version", Topology.SIMPLICITY_VERSION);`

should change to something like:

`properties.put("simplicity.version", "2.0");`

and the import of Topology removed.

If WebSphere Liberty requires some additional classes then those classes should only be added to the WebSphere Liberty side so as not to pollute Open Liberty's source.

**Comment by jhanders34** [comment]
It looks like there are some additional changes needed to clean up dead code in some of the FAT projects.  I hadn't updated some of the settings in fattest.simplicity bnd when I originally ran with the changes outlined in the PR review.

Additional changes that are needed:

- com.ibm.ws.logging.htpel.fat.CommonTasks, need to remove `void removeUnusedServers(Cell cell, Set<Server> excludedServers) throws Exception`, `void remove(Server server) throws Exception` and `void startServer(Server serverToStart) throws Exception` methods.  The imports will also need to be removed that are no longer used with the removal of those methods.

- io.openliberty.wsoc.util.WebServerControl, add:
```
    enum ProcessStatus {
        RUNNING, STOPPED
    };
```
and remove the `com.ibm.websphere.simplicity.runtime.ProcessStatus` import.

Additionally it looks like:

```
com.ibm.ws.componenttest:com.ibm.componenttest.common:1.0.0
com.ibm.ws.componenttest:com.ibm.ws.topology.helper:1.0.0
com.ibm.ws.componenttest:fat.harness:1.0.0
```
are not needed and `com.ibm.ws.componenttest:fat.util:1.0.0` only has 4 classes so it would make sense to add the source for those 4 classes as part of this PR as well.  It looks like the UnitConverter method may not be used so its source may not need to be added.

**Comment by tevans78** [comment]
#build

**Review by jhanders34 — APPROVED** [review]
I am marking it approved.  I am not sure why there are copyright updates for the com.ibm.websphere.simplicity.config files in this PR, but I can live with them.  I do agree that a squash should be done of at least the first 3 commits in the PR.

**Comment by tevans78** [comment]
@jhanders34 I have now squashed, rebased and removed those extra copyright changes from the config package.
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
Finally got a green PB so good to merge

**Comment by tevans78** [comment]
#libby

**Pull request merged**